### PR TITLE
Add support for specifying pid namespace by container id

### PR DESCRIPTION
**Pull request by mrunalp**
This is for adding support for https://github.com/docker/docker/issues/22479 to docker.

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
@calavera @cpuguy83 PTAL

**Comment by calavera** [comment]
LGTM

**Comment by MHBauer** [comment]
Is there an explanation for the term "pid stack"? Is it the same as "pid namespace"? 

It'd be good to have a unit test, but there aren't any to start with, so I understand why there aren't new ones.

**Comment by mrunalp** [comment]
@MHBauer pid stack is the same as pid namespace. (I am following comments in existing code).
There are unit tests for this in the docker repo. I'll add those in my docker PR.

File: types/container/host_config.go [hunk 1]
```diff
@@ -141,25 +141,44 @@ type PidMode string
...
-	return !(n.IsHost())
+	return !(n.IsHost() || n.IsContainer())
 }
 
 // IsHost indicates whether the container uses the host's pid stack.
 func (n PidMode) IsHost() bool {
 	return n == "host"
 }
 
+// IsContainer indicates whether the container uses a container's pid stack.
```
Comment (LK4D4) [inline types/container/host_config.go]: I'm pretty sure that stack is from "network stack" :) I don't think that it's applicable here.

**Comment by mrunalp** [comment]
@LK4D4 Changed to pid namespace :)

**Comment by LK4D4** [comment]
@mrunalp thanks, LGTM

**Comment by LK4D4** [comment]
@mrunalp it's vendor time!

**Pull request merged**